dpdispatcher/lsf.py

```diff
@@ -2,141 +2,41 @@
-#             prepend_text="",
-#             append_text="",
-#             gpu_usage=True,
-#             gpu_new_syntax=True,
-#             lsf_bsub_dict=None,
-#             group_size=1
-#     ):
-#         """define LSF resources
-#
-#         Parameters
-#         ----------
-#         number_node: nodes to be used
-#         cpu_per_node: CPU cores used on each node
-#         gpu_per_node: GPU
-#         queue_name: the name of queue
-#         walltime: max time of task
-#         prepend_text: prepend scripts, code executed before the task run
-#         append_text: append scripts, code executed after the task run
-#         gpu_usage: choose if GPU line is used
-#         lsf_bsub_dict: other bsub parameters.
-#         group_size: tasks contained by each group
-#         """
-#         super().__init__(number_node, cpu_per_node, gpu_per_node, queue_name, group_size)
-#         if lsf_bsub_dict is None:
-#             lsf_bsub_dict = {}
-#         self.walltime = walltime
-#         self.gpu_new_syntax = gpu_new_syntax
-#         self.gpu_usage = gpu_usage
-#         self.prepend_text = prepend_text
-#         self.append_text = append_text
-#         self.lsf_bsub_dict = lsf_bsub_dict
 
-
-class LSF(Machine):
+class LSF(Machine, ABC):
     """
     LSF batch
     """
     def gen_script(self, job):
-        # if type(job.resources) is LSFResources:
-        #     resources = job.resources
-        #     lsf_bsub_dict = job.resources.lsf_bsub_dict
-        #     if lsf_bsub_dict is None:
-        #         lsf_bsub_dict = {}
-        # elif type(job.resources) is Resources:
-        #     resources = LSFResources(**job.resources.__dict__)
-        #     lsf_bsub_dict = {}
-        # else:
-        #     raise RuntimeError('type job.resource error')
+        lsf_script = super(LSF, self).gen_script(job)
+        return lsf_script
 
+    def gen_script_header(self, job):
         resources = job.resources
-        lsf_bsub_dict = resources.extra_specification.copy()
-
-        # headers
         script_header_dict = {
             'lsf_nodes_line': "#BSUB -n {number_cores}".format(
                 number_cores=resources.number_node * resources.cpu_per_node),
             'lsf_ptile_line': "#BSUB -R 'span[ptile={cpu_per_node}]'".format(
                 cpu_per_node=resources.cpu_per_node),
             'lsf_partition_line': "#BSUB -q {queue_name}".format(
-                queue_name=resources.queue_name),
-            'lsf_walltime_line': "#BSUB -W {walltime}".format(
-                walltime=resources.kwargs.get('walltime', '12:00'))
+                queue_name=resources.queue_name)
         }
-        gpu_usage_flag = resources.kwargs.get('gpu_usage', False)
-        gpu_new_syntax_flag = resources.kwargs.get('gpu_new_syntax', False)
+        gpu_usage_flag = resources.kwargs['kwargs'].get('gpu_usage', False)
+        gpu_new_syntax_flag = resources.kwargs['kwargs'].get('gpu_new_syntax', False)
```

Comment:
We don't have documents for these two parameters. Maybe we need to discuss where to add documents first.